app/services/task_service/gpt_generate_tests.rb

```diff
@@ -37,13 +57,14 @@ def gpt_response
       )
 
       # parse out the response
-      response.dig('choices', 0, 'message', 'content')
-    end
-
-    private
+      raw_response = response.dig('choices', 0, 'message', 'content')
 
-    def client
-      @client ||= OpenAI::Client.new
+      # check for ``` in the response and extract the text between the first set
+      if raw_response.include?('```')
+        raw_response[/```(.*?)```/m, 1].lines[1..]&.join&.strip
+      else
+        raw_response
+      end
```

Comment:
I am eager to hear Karol's opinion here, but one could consider extracting this part to a dedicated method. Nothing required from my side, just optional.

Reply:
I don't see the need to extract that part, but would refactor the if-else into a guard-clause. (since the else-branch now contains a raise)